go/workflow-interceptor-updates.md

```diff
@@ -0,0 +1,269 @@
+# Go SDK Workflow Interceptor Updates and OpenTelemetry Tracing Support
+
+This proposal is for updates to Go workflow interceptors with the following features:
+
+* Support for and implementation of OpenTelemetry tracing
+* Support idiomatic Go interceptor design
+* Provide feature parity with Java SDK and other missing features (see https://github.com/temporalio/sdk-go/issues/529)
+* Add activity interceptors
+* Make inbound interceptors future-proof for argument addition
+
+**These updates are known to NOT be backwards-compatible**
+
+## Interceptor Proposed Updates
+
+### WorkflowInterceptor.InterceptWorkflow
+
+Current:
+
+```go
+InterceptWorkflow(info *WorkflowInfo, next WorkflowInboundCallsInterceptor) WorkflowInboundCallsInterceptor
+```
+
+Proposal:
+
+```go
+InterceptWorkflow(ctx Context, next WorkflowInboundCallsInterceptor) WorkflowInboundCallsInterceptor
+```
+
+Why: Callers can use the context to get workflow info.
+
+### WorkflowXCallsInterceptor
+
+Proposal: Change the name to `WorkflowXInterceptor`. Why: `Calls` just adds noise (more accepted in Java, not Go).
+
+### WorkflowInboundCallsInterceptor.ExecuteWorkflow
```

Comment:
Isn't this supposed to be `WorkflowInboundInterceptor` according to the line above?

Reply:
Note, for clarity I am changing these headers to desired names instead of what they were.